src/io/flutter/view/FlutterView.java

```diff
@@ -0,0 +1,250 @@
+    toolbarGroup.add(new RepaintRainbowAction(this));
+    toolbarGroup.add(new PerformanceOverlayAction(this));
+    toolbarGroup.add(new TimeDilationAction(this));
+    toolbarGroup.addSeparator();
+    toolbarGroup.add(new ObservatoryTimelineAction(this));
+
+    final JPanel mainContent = new JPanel(new BorderLayout());
+
+    toolWindowPanel.setToolbar(ActionManager.getInstance().createActionToolbar("FlutterViewToolbar", toolbarGroup, true).getComponent());
+    toolWindowPanel.setContent(mainContent);
+
+    final ContentManager contentManager = toolWindow.getContentManager();
+    contentManager.addContent(toolContent);
+    contentManager.setSelectedContent(toolContent);
+  }
+
+  /**
+   * Called when a debug connection starts.
+   */
+  public void debugActive(FlutterViewMessages.FlutterDebugEvent event) {
+    this.debugEvent = event;
+
+    event.vmService.addVmServiceListener(new VmServiceListener() {
+      @Override
+      public void connectionOpened() {
+      }
+
+      @Override
+      public void received(String s, Event event) {
+      }
+
+      @Override
+      public void connectionClosed() {
+        FlutterView.this.debugEvent = null;
+      }
+    });
+  }
+
+  FlutterApp getFlutterApp() {
+    if (debugEvent == null) {
+      return null;
+    }
+    return debugEvent.observatoryConnector.getApp();
+  }
+
+  /**
+   * State for the view.
+   */
+  class State {
+    State() {
+    }
+  }
+}
+
+abstract class AbstractToggleableAction extends DumbAwareAction implements Toggleable {
+  @NotNull final FlutterView view;
+  private boolean selected = false;
+
+  AbstractToggleableAction(@NotNull FlutterView view, @Nullable String text, @Nullable String description, @Nullable Icon icon) {
+    super(text, description, icon);
+
+    this.view = view;
+  }
+
+  @Override
+  public final void update(AnActionEvent e) {
+    // enabled
+    e.getPresentation().setEnabled(view.getFlutterApp() != null);
+
+    // selected
+    final boolean selected = this.isSelected(e);
+    final Presentation presentation = e.getPresentation();
+    presentation.putClientProperty("selected", selected);
+  }
+
+  @Override
+  public void actionPerformed(AnActionEvent event) {
+    this.setSelected(event, !isSelected(event));
+    final Presentation presentation = event.getPresentation();
+    presentation.putClientProperty("selected", isSelected(event));
+
+    FlutterInitializer.sendActionEvent(this);
+    perform(event);
+  }
+
+  protected abstract void perform(AnActionEvent event);
+
+  public boolean isSelected(AnActionEvent var1) {
+    return selected;
+  }
+
+  public void setSelected(AnActionEvent event, boolean selected) {
+    this.selected = selected;
+
+    ApplicationManager.getApplication().invokeLater(() -> this.update(event));
+  }
+}
+
+class DebugDrawAction extends AbstractToggleableAction {
+  DebugDrawAction(@NotNull FlutterView view) {
+    super(view, "Toggle Debug Paint", "Toggle Debug Painting", AllIcons.General.TbShown);
```

Comment:
In general I think we've been externalizing strings like these.  (Not wedded to that though!)